table/table_test.go

```diff
@@ -280,10 +280,10 @@ func (t *TableWritingTestSuite) writeParquet(fio iceio.WriteFileIO, filePath str
 
 func (t *TableWritingTestSuite) createTable(identifier table.Identifier, formatVersion int, spec iceberg.PartitionSpec, sc *iceberg.Schema) *table.Table {
 	meta, err := table.NewMetadata(sc, &spec, table.UnsortedSortOrder,
-		t.getMetadataLoc(), iceberg.Properties{"format-version": strconv.Itoa(formatVersion)})
+		t.location, iceberg.Properties{"format-version": strconv.Itoa(formatVersion)})
```

Comment:
👍  table's base location (`t.Location()`) is actually the table metadata's location (`t.metadata.Location()`).

https://github.com/apache/iceberg-go/blob/5142432b98b90b33c3f6af417de4d40aa3be8946/table/table.go#L57-L64

Reply:
Yea, it's a little weird, but I get the handle on it now. the metadata only knows the table's base location (`t.metadata.Location()`) while the table itself knows the actual location of the metadata json file (`t.metadataLocation`).